lib/string_reverse.rb

```diff
@@ -1,6 +1,18 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is number of characters in the string

```

Comment:
Why?

Reply:
Here's how I answered these. This is about as thorough as I can think to be in analyzing space & time complexity: https://github.com/mmcknett/string_reverse/blob/implementation/lib/string_reverse.rb#L3-L11